Assets/BowAndArrowImproved/Scripts/HealthBar.cs

```diff
@@ -9,7 +9,8 @@ public class HealthBar : MonoBehaviour
     [SerializeField] private GameObject healthSquare;
     
     [SerializeField] private HitCountHittable hittable;
-
+    [SerializeField] private int health;

```

Comment:
this is a UI class, it only represents the UI of the health bar, it shouldn't hold the health count, only HitCountHittable, no one else!